modules/ztp-image-based-upgrade-generate-seed-image.adoc

```diff
@@ -0,0 +1,267 @@
+<2> Specifies a base64-encoded authfile for write-access to the registry for pushing the generated seed images.
+<3> Optional. Specifies a base64-encoded `KUBECONFIG` file for administrator access to the hub cluster to unregister the seed cluster from {rh-rhacm}. If this is not present in the secret, the {rh-rhacm} cleanup is skipped.
+--
+
+.. Create the authentication and the `Kubeconfig` files by running the following commands:
++
+--
+.Authentication file
+[source,terminal]
+----
+$ MY_USER=myuserid
+$ AUTHFILE=/tmp/my-auth.json
+$ podman login --authfile ${AUTHFILE} -u ${MY_USER} quay.io/${MY_USER}
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${AUTHFILE} ; echo
+----
+
+.hubKubeconfig file
+[source,terminal]
+----
+$ HUBKUBECONFIGFILE=/home/<user_ID>/hubkubeconfig
+----
+
+[source,terminal]
+----
+$ base64 -w 0 ${HUBKUBECONFIGFILE} ; echo
+----
+--
+
+.. Copy the outputs into the respective `seedAuth` and `hubKubeconfig` fields in the `Secret` CR.
+
+. Apply the `Secret` CR.
++
+[source,terminal]
+----
+$ oc apply -f secretseedgenerator.yaml
+----
+
+. (Optional) If you did not populate the `hubKubeconfig` field in the `Secret` CR, detach the seed cluster from the hub cluster and remove all {rh-rhacm} and multicluster engine certificates that cannot be recertificated in the target cluster.
+
+.. Detach the seed cluster from the hub cluster:
++
+[source,terminal]
+----
+$ oc delete managedcluster sno-worker-00
+----
+
+.. Wait until the `ManagedCluster` CR is removed. Once the CR is removed and create the proper `SeedGenerator` CR. The {lcao} cleans up the {rh-rhacm} artifacts.
+
+. Create the `SeedGenerator` CR:
++
+--
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: SeedGenerator
+metadata:
+  name: seedimage <1>
+spec:
+  seedImage: <seed_container_image> <2>
+  recertImage: "quay.io/edge-infrastructure/recert:latest"
+----
+<1> The `SeedGenerator` CR must be named `seedimage`.
+<2> Specify the container image URL, for example, `quay.io/example/seed-container-image:<tag>`. It is recommended to use the `<seed_cluster_name>:<ocp_version>` format.
+
+
+[IMPORTANT]
+====
+The same version of the applications must function on both the current and the target release of {product-title}.
+====
+--
+
+. Generate the seed image by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f seedgenerator.yaml
+----
+
++
+[IMPORTANT]
+====
+The cluster reboots and loses API capabilities while the {lcao} generates the seed image.
+====
+
++
+Applying the `SeedGenerator` CR stops the `kubelet` and the CRI-O operations, and then it starts the image generation.
+Finally, the {lcao} starts a cleanup process that allows you to keep the seed {sno} fully functional.
+//This might not be important info, check
+
+.Verification
+
+. You can monitor the image generation process.
+// this sounds all kind of wrong, check where you can monitor it exactly
+
+.. SSH into the seed cluster:
++
+[source,terminal]
+----
+$ ssh <node_ID>
```

Comment:
ssh core@<node_ID>

Not sure if this is more troubleshooting than verification. I mean, verification must be led by the status of the seedgenerator CR. Since the cluster is not available during the image creation process, the user should wait for the cluster to be up and running again. In case the cluster is not responsive after some time, we could check if there was any issue during the image creation process by connecting as shown in the example.  Wdyt?

Reply:
I chatted with someone about creating a dedicated Troubleshooting section, I think that would be cleaner. Once we established some common scenarios, I'm happy to create that section, so I can put this there and expand depending on the issue. For now, I'm going to comment it out and add a TODO for myself. Keeping the CR monitoring step in Verification